embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/logs/LogOrchestrator.kt

```diff
@@ -0,0 +1,74 @@
+package io.embrace.android.embracesdk.internal.logs
+
+import io.embrace.android.embracesdk.comms.delivery.DeliveryService
+import io.embrace.android.embracesdk.internal.clock.Clock
+import io.embrace.android.embracesdk.worker.ScheduledWorker
+import java.util.concurrent.ScheduledFuture
+import java.util.concurrent.TimeUnit
+
+internal class LogOrchestrator(
+    private val logOrchestratorScheduledWorker: ScheduledWorker,
+    private val clock: Clock,
+    private val sink: LogSink,
+    private val deliveryService: DeliveryService
+) {
+    private var lastLogTime: Long = 0
```

Comment:
These should be volatile or be Atomic references since they are accesses from multiple threads?

Reply:
Set to volatile